packages/checkout-storefront/src/sections/PaymentSection/AdyenDropIn.tsx

```diff
@@ -0,0 +1,193 @@
+) => Promise<void> | void;
+
+export const AdyenDropIn = memo<AdyenDropInProps>(({}) => {
+  const {
+    env: { checkoutApiUrl },
+  } = useAppConfig();
+
+  const dropinContainerElRef = useRef<HTMLDivElement>(null);
+  const dropinComponentRef = useRef<DropinElement | null>(null);
+
+  const [adyenCheckoutInstance, setAdyenCheckoutInstance] = useState<AdyenCheckoutInstance | null>(
+    null
+  );
+
+  const { checkout, loading: isCheckoutLoading } = useCheckout();
+  const getCheckout = () => checkout;
+
+  const [adyenSessionResponse] = useFetch(createDropInAdyenSession, {
+    args: {
+      checkoutApiUrl,
+      checkoutId: checkout?.id,
+      // we send 0 here and update it later inside `onSubmit`
+      totalAmount: 0,
+      currency: checkout?.totalPrice?.gross?.currency,
+      provider: "adyen",
+      method: "dropin",
+      redirectUrl: window.location.href,
+    },
+    skip: isCheckoutLoading,
+  });
+
+  const [, fetchCreateDropInAdyenPayment] = useFetch(createDropInAdyenPayment, {
+    skip: true,
+  });
+
+  const onSubmit: AdyenCheckoutInstanceOnSubmit = useEvent(async (state, component) => {
+    const result = await fetchCreateDropInAdyenPayment({
+      checkoutApiUrl,
+      totalAmount: checkout.totalPrice.gross.amount,
+      checkoutId: checkout.id,
+      method: "dropin",
+      provider: "adyen",
+      redirectUrl: window.location.href,
+      adyenStateData: state.data,
+    });
+
+    if (!result || "message" in result) {
+      // @todo handle error ?
+      console.error(result);
+      return;
+    }
+    if (result?.payment.action) {
+      // @todo Drop-in will handle the required action, depending on the action.type.
+      // You will then need to make one more API call to verify the payment result.
+      return;
+    } else {
+      switch (result.payment.resultCode) {
+        // @todo https://docs.adyen.com/online-payments/payment-result-codes
+        case AdyenPaymentResponse.ResultCodeEnum.AuthenticationFinished:
+        case AdyenPaymentResponse.ResultCodeEnum.Cancelled:
+        case AdyenPaymentResponse.ResultCodeEnum.ChallengeShopper:
+        case AdyenPaymentResponse.ResultCodeEnum.Error:
+        case AdyenPaymentResponse.ResultCodeEnum.IdentifyShopper:
+        case AdyenPaymentResponse.ResultCodeEnum.Pending:
+        case AdyenPaymentResponse.ResultCodeEnum.PresentToShopper:
+        case AdyenPaymentResponse.ResultCodeEnum.Received:
+        case AdyenPaymentResponse.ResultCodeEnum.RedirectShopper:
+        case AdyenPaymentResponse.ResultCodeEnum.Refused: {
+          console.error(result);
+          component.setStatus("error", {
+            message: `${result.payment.resultCode}: ${result.payment.refusalReason}`,
+          });
+          return;
+        }
+
+        case AdyenPaymentResponse.ResultCodeEnum.Authorised:
+        case AdyenPaymentResponse.ResultCodeEnum.Success: {
+          const newUrl = `?order=${result.orderId}`;
+          window.location.href = newUrl;
```

Comment:
I think we could abstract the logic of getting URL to to `OrderDetails` page. There's the same logic in `src/hooks/usePay.ts:48`